browser/brave_content_browser_client.cc

```diff
@@ -430,7 +430,9 @@ void BraveContentBrowserClient::RegisterBrowserInterfaceBindersForFrame(
         base::BindRepeating(&BindBraveAdsHost));
   }
 
-  if (brave_wallet::IsNativeWalletEnabled()) {
+  if (brave_wallet::IsNativeWalletEnabled() &&
+      brave_wallet::IsAllowedForContext(
```

Comment:
FYI: This check is redundant but it doesn't hurt. `MaybeBindBraveWalletProvider` will fail already for all the factories and do nothing in the same condition.   I think leaving it is better because it makes it more explicit what happens here.